google/cloud/bigquery/job/query.py

```diff
@@ -1300,12 +1300,47 @@ def result(
                 If the job did not complete in the given timeout.
         """
         try:
-            super(QueryJob, self).result(retry=retry, timeout=timeout)
+            retry_do_query = getattr(self, "_retry_do_query", None)
+            first = True
```

Comment:
`first` might be unnecessary if we use a `retry_query` object here that catches the exceptions for failed jobs.

https://github.com/googleapis/python-api-core/blob/d566f2893eb708c109ebc3b4fa16f57aa4c65f26/google/api_core/retry.py#L257

Does `super(QueryJob, self).result(retry=retry, timeout=timeout)` raise if the job is already failed from the initial POST request (`_begin`)? I recall we have some logic not to reload the job state / call getQueryResults if the job state is already DONE. -- If not, maybe we should do that, since we document it as such and it'd be weird to be inconsistent with how we raise depending on when the job failed.